### DISCUSSION TITLE
How to Add Child Nodes to the Palette in Sirius?

### DISCUSSION THREAD

**Discussion by imeneazouzi** [eclipse-sirius/sirius-components#3573] [Q&A]
I have created a metamodel that contains activities, and within each activity, I can create sub-activities. However, in the sub-activity palette, there is no option to add sub-sub-activities to an activity, even though I can do this in the Sirius desktop version. I have tried modifying the GroupPalette code, but it doesn't work. Can you tell me which component is responsible for adding the possibility of creating additional child nodes in the palette?

**Comment by pcdavid** [comment]
The content of the palette displayed when you select a node (resp. the diagram's background) is defined in the "Palette" element of the corresponding NodeDescription (resp. DiagramDescription) in your View model:

![screenshot-2024-06-12_09-08](https://github.com/eclipse-sirius/sirius-web/assets/10608/94ce16d1-6073-4758-8363-885f624dba6e)

If you want a tool creating a sub-activity to appear in the palette of your activity nodes, simply add a new "Node Tool" inside the node description's palette, and like in Sirius Desktop, create the sub-activity semantic element in the body of the tool. If your node description is synchronized (the default) and you have a corresponding (sub-)node mapping to display sub-activities, the corresponding node will appear automatically.
